DevCycle/DevCycleUser.swift

```diff
@@ -74,14 +69,24 @@ public class UserBuilder {
     }
 
     public func build() throws -> DevCycleUser {
-        guard self.user.userId?.trimmingCharacters(in: .whitespacesAndNewlines) != ""
-        else {
-            throw UserError.InvalidUser
+        // First, clean and validate the userId
+        let trimmedUserId = self.user.userId?.trimmingCharacters(in: .whitespacesAndNewlines)
```

Comment:
why? technically whitespace and newlines are valid string characters that can affect bucketing? Which other SDKs do we do this in?

Reply:
yea I think you are correct, I don't see us trimming anywhere else.